src/observers/link_prefetch_observer.js

```diff
@@ -0,0 +1,184 @@
+import {
+  doesNotTargetIFrame,
+  getLocationForLink,
+  getMetaContent,
+  findClosestRecursively
+} from "../util"
+import { prefetchCache, cacheTtl } from "../core/drive/prefetch_cache"
+
+export class LinkPrefetchObserver {
+  started = false
+  delayBeforePrefetching = 100
+  hoverTriggerEvent = "mouseenter"
+  touchTriggerEvent = "touchstart"
```

Comment:
In addition to mouse and touch events, should the observer also monitor [focusin](https://developer.mozilla.org/en-US/docs/Web/API/Element/focusin_event) events for keyboard navigators to also benefit from pre-fetching?

Reply:
This would trigger requests all the time when pressing tab, right? Or by browsing the page with the keyboard as a screen reader user does. I think it could be a machine gun of requests in that case.

However, if the focusin event is not considered, screen reader users cannot take advantage of this feature at all.

Reply:
That's a great idea, as @afcapel noted, let's add that on a follow-up PR!

Reply:
@brunoprietog The delay feature would prevent this. 

i.e. It would prevent a flurry of requests from happening if the user is fast navigating links using the Tab key. However, once the user stops at a particular link, we'd prefetch it.

Reply:
I'm not so sure that value is satisfactory enough, as an average screen reader user navigates more slowly. This is because when focusing on a control, you have to listen to what it contains before making a decision. It could even be seconds. I really think it's more complicated than I thought.